Run dubbo demo FooProviderBootstrap fail

## Issue Description

Type: *bug report* 

### Describe what happened (or what feature you want)
run FooProviderBootstrap.main() fail

Exception in thread "main" java.lang.NoClassDefFoundError: com/alibaba/spring/beans/factory/annotation/AnnotationInjectedBeanPostProcessor
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.alibaba.dubbo.config.spring.context.annotation.DubboComponentScanRegistrar.registerReferenceAnnotationBeanPostProcessor(DubboComponentScanRegistrar.java:90)
	at com.alibaba.dubbo.config.spring.context.annotation.DubboComponentScanRegistrar.registerBeanDefinitions(DubboComponentScanRegistrar.java:61)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.lambda$loadBeanDefinitionsFromRegistrars$1(ConfigurationClassBeanDefinitionReader.java:358)
	at java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsFromRegistrars(ConfigurationClassBeanDefinitionReader.java:357)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:145)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:117)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:328)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:233)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:273)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:93)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:694)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:532)
	at com.alibaba.csp.sentinel.demo.dubbo.demo1.FooProviderBootstrap.main(FooProviderBootstrap.java:50)
Caused by: java.lang.ClassNotFoundException: com.alibaba.spring.beans.factory.annotation.AnnotationInjectedBeanPostProcessor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 26 more
### Describe what you expected to happen


### How to reproduce it (as minimally and precisely as possible)
1. check out master
2. run com.alibaba.csp.sentinel.demo.dubbo.demo1.FooProviderBootstrap.main()

### Tell us your environment
branch: master


sentinel-demo-dubbo dubbo version is 2.6.5  class ReferenceAnnotationBeanPostProcessor extends AnnotationInjectedBeanPostProcessor.
When I change dubbo version to 2.5.8 it is ok .
class ReferenceAnnotationBeanPostProcessor extends InstantiationAwareBeanPostProcessorAdapter

--- a/sentinel-demo/sentinel-demo-dubbo/pom.xml
+++ b/sentinel-demo/sentinel-demo-dubbo/pom.xml
@@ -13,14 +13,19 @@
 
     <dependencies>
         <dependency>
-            <groupId>org.springframework</groupId>
-            <artifactId>spring-context</artifactId>
-            <version>5.0.7.RELEASE</version>
-        </dependency>
-        <dependency>
             <groupId>com.alibaba</groupId>
             <artifactId>dubbo</artifactId>
             <version>2.6.5</version>
+        </dependency>
+        <dependency>
+            <groupId>io.netty</groupId>
+            <artifactId>netty-all</artifactId>
+            <version>4.1.31.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>com.alibaba.spring</groupId>
+            <artifactId>spring-context-support</artifactId>
+            <version>1.0.2</version>
         </dependency>
         <dependency>
             <groupId>com.alibaba.csp</groupId>

--- a/sentinel-demo/sentinel-demo-dubbo/pom.xml
+++ b/sentinel-demo/sentinel-demo-dubbo/pom.xml
@@ -28,8 +28,8 @@
             <version>4.1.31.Final</version>
         </dependency>
 
-        <!-- As demo use @DubboComponentScan to config, which is implemented in dubbo-config module -->
-        <!-- The dubbo-config module is optional and it depends spring-context-support, so add it explicitly-->
+        <!-- As demo use @DubboComponentScan to config, which is implemented in dubbo-config-spring module -->
+        <!-- The dubbo-config-spring module is optional and it depends spring-context-support, so add it explicitly-->
         <!-- @see https://github.com/apache/incubator-dubbo/issues/2869 -->
         <dependency>
             <groupId>com.alibaba.spring</groupId>

--- a/sentinel-demo/sentinel-demo-dubbo/pom.xml
+++ b/sentinel-demo/sentinel-demo-dubbo/pom.xml
@@ -12,21 +12,32 @@
     <artifactId>sentinel-demo-dubbo</artifactId>
 
     <dependencies>
+        <!-- Demo use dubbo 2.6.5, since it supports jdk1.6 -->
         <dependency>
             <groupId>com.alibaba</groupId>
             <artifactId>dubbo</artifactId>
             <version>2.6.5</version>
         </dependency>
+
+        <!-- As dubbo provide qos plugin since 2.5.8 and is enable by default -->
+        <!-- The dubbo-qos module is optional and it depends netty4, so add it explicitly -->
+        <!-- @see http://dubbo.apache.org/zh-cn/docs/user/references/qos.html -->
         <dependency>
             <groupId>io.netty</groupId>
             <artifactId>netty-all</artifactId>
             <version>4.1.31.Final</version>
         </dependency>
+
+        <!-- As demo use @DubboComponentScan to config, which is implemented in dubbo-config module -->
+        <!-- The dubbo-config module is optional and it depends spring-context-support, so add it explicitly-->
+        <!-- @see https://github.com/apache/incubator-dubbo/issues/2869 -->
         <dependency>
             <groupId>com.alibaba.spring</groupId>
             <artifactId>spring-context-support</artifactId>
             <version>1.0.2</version>
         </dependency>
+
+        <!-- sentinel adapter and transport -->
         <dependency>
             <groupId>com.alibaba.csp</groupId>
             <artifactId>sentinel-dubbo-adapter</artifactId>

